sw/airborne/firmwares/rotorcraft/stabilization/stabilization_attitude_heli_indi.c

```diff
@@ -0,0 +1,656 @@
+#define STABILIZATION_ATTITUDE_HELI_INDI_NOTCHFILT_BW_THRUST STABILIZATION_ATTITUDE_HELI_INDI_NOTCHFILT_BW_DEFAULT
+#endif
+#ifndef STABILIZATION_ATTITUDE_HELI_INDI_BUTTERW_CUTOFF_DEFAULT
+#define STABILIZATION_ATTITUDE_HELI_INDI_BUTTERW_CUTOFF_DEFAULT 40.0
+#endif
+#ifndef STABILIZATION_ATTITUDE_HELI_INDI_BUTTERW_CUTOFF_ROLL
+#define STABILIZATION_ATTITUDE_HELI_INDI_BUTTERW_CUTOFF_ROLL STABILIZATION_ATTITUDE_HELI_INDI_BUTTERW_CUTOFF_DEFAULT
+#endif
+#ifndef STABILIZATION_ATTITUDE_HELI_INDI_BUTTERW_CUTOFF_PITCH
+#define STABILIZATION_ATTITUDE_HELI_INDI_BUTTERW_CUTOFF_PITCH STABILIZATION_ATTITUDE_HELI_INDI_BUTTERW_CUTOFF_DEFAULT
+#endif
+#ifndef STABILIZATION_ATTITUDE_HELI_INDI_BUTTERW_CUTOFF_YAW
+#define STABILIZATION_ATTITUDE_HELI_INDI_BUTTERW_CUTOFF_YAW STABILIZATION_ATTITUDE_HELI_INDI_BUTTERW_CUTOFF_DEFAULT
+#endif
+#ifndef STABILIZATION_ATTITUDE_HELI_INDI_BUTTERW_CUTOFF_THRUST
+#define STABILIZATION_ATTITUDE_HELI_INDI_BUTTERW_CUTOFF_THRUST STABILIZATION_ATTITUDE_HELI_INDI_BUTTERW_CUTOFF_DEFAULT
+#endif
+#ifndef STABILIZATION_ATTITUDE_HELI_INDI_NOTCH_MIN_RPM
+#define STABILIZATION_ATTITUDE_HELI_INDI_NOTCH_MIN_RPM 1500
+#endif
+#define INDI_NOTCH_MIN_RPM STABILIZATION_ATTITUDE_HELI_INDI_NOTCH_MIN_RPM
+
+/* Shorter defines to use lateron in the matrix */
+#define INVG_00 STABILIZATION_ATTITUDE_HELI_INDI_GINV_ROLL_TO_ROLL
+#define INVG_11 STABILIZATION_ATTITUDE_HELI_INDI_GINV_PITCH_TO_PITCH
+#define INVG_22 STABILIZATION_ATTITUDE_HELI_INDI_GINV_YAW_TO_YAW
+#define INVG_33 -50000 // Not used at the moment
+
+struct Int32Quat   stab_att_sp_quat;
+struct Int32Eulers stab_att_sp_euler;
+struct Int32Quat sp_offset; // non-zero neutral attitude
+
+struct HeliIndiGains heli_indi_gains = {
+  STABILIZATION_ATTITUDE_HELI_INDI_ROLL_P,
+  STABILIZATION_ATTITUDE_HELI_INDI_PITCH_P,
+  STABILIZATION_ATTITUDE_HELI_INDI_YAW_P,
+  STABILIZATION_ATTITUDE_HELI_INDI_YAW_D
+};
+
+/* Main controller struct */
+struct IndiController_int heli_indi_ctl;
+
+/* Filter functions */
+struct delayed_first_order_lowpass_filter_t actuator_model[INDI_DOF];
+int32_t alpha_yaw_inc; // Tail model parameters for spinning up
+int32_t alpha_yaw_dec; // Tail model parameters for spinning down
+/* Measurement filters */
+Butterworth2LowPass_int actuator_lowpass_filters[INDI_DOF];
+Butterworth2LowPass_int measurement_lowpass_filters[INDI_DOF];
+struct SecondOrderNotchFilter actuator_notchfilter[INDI_DOF];
+struct SecondOrderNotchFilter measurement_notchfilter[INDI_DOF];
+#if STABILIZATION_ATTITUDE_HELI_INDI_USE_FAST_DYN_FILTERS
+struct delayed_first_order_lowpass_filter_t fast_dynamics_model[2]; // only pitch and roll
+#endif
+
+#if PERIODIC_TELEMETRY
+#include "subsystems/datalink/telemetry.h"
+
+/* Telemetry messages here, at the moment there are none */
+
+#endif // PERIODIC_TELEMETRY
+
+static inline void indi_subtract_vect(int32_t _out[], int32_t _in1[], int32_t _in2[])
+{
+  for (uint8_t i = 0; i < INDI_DOF; i++) {
+    _out[i] = _in1[i] - _in2[i];
+  }
+}
+
```

Comment:
There seems to be some functions that have this functionally, check sw/airborne/math/pprz_simple_matrix.h. 

If you have time, you could add a math file which has the wrappers for the matrix computations for vectors. I think these functions would be interesting for others not just for indi, you would just make the INDI_DOF the input t the general function. 

If you don't have time (or the interest) maybe make an issue for to do this.